packages/code-signing-certificates/src/index.ts

```diff
@@ -0,0 +1,144 @@
+import assert from 'assert';
+import { md, pki as PKI, random, util } from 'node-forge';
+
+// a hexString is considered negative if it's most significant bit is 1
+// because serial numbers use ones' complement notation
+// this RFC in section 4.1.2.2 requires serial numbers to be positive
+// http://www.ietf.org/rfc/rfc5280.txt
+function toPositiveHex(hexString: string) {
+  let mostSiginficativeHexAsInt = parseInt(hexString[0], 16);
+  if (mostSiginficativeHexAsInt < 8) {
+    return hexString;
+  }
+  mostSiginficativeHexAsInt -= 8;
```

Comment:
```suggestion
  if (mostSiginficativeHexAsInt > 8) {
    return hexString;
  }
  mostSiginficativeHexAsInt += 8;
```

otherwise wouldn't the case of `9` return a negative hex?

Reply:
Anything less than hex 8 (8) has a most significant digit of 0 when represented in binary (verified in https://www.rapidtables.com/convert/number/hex-to-binary.html), so if the most significant digit is 0-8, the number is fine and it should be returned verbatim. When it is greater than 8, we subtract 8 from the first hex digit (making it be in the range 0-8) so that the number has a 0 most significant digit when represented in binary.